### PR TITLE
Fix pc_patch_compress leak (take 2)

### PR DESCRIPTION
**Pull request by elemoine**
This PR replaces https://github.com/pgpointcloud/pointcloud/pull/112.

I've taken @achidlow's commits, squashed them together, and added three additional commits. My second commit renames the pc_patch_common_free function to pc_patch_free_stats, which is more explicit.

@mbredif can you please take a final look at this before I merge it and release 1.1.0?

### CODE REVIEW

**Comment by elemoine** [comment]
Closes #111.

File: lib/pc_patch_uncompressed.c [hunk 1]
```diff
@@ -215,6 +216,8 @@ pc_patch_uncompressed_compute_extent(PCPATCH_UNCOMPRESSED *patch)
 void
 pc_patch_uncompressed_free(PCPATCH_UNCOMPRESSED *patch)
 {
+	pc_patch_free_stats((PCPATCH*) patch);
+
```
Comment (mbredif) [inline lib/pc_patch_uncompressed.c]: As you removed the check in `pc_patch_free_stats`, maybe you can add `assert(patch);` here, just to be consistent across patch types

Comment (elemoine) [reply lib/pc_patch_uncompressed.c]: Good point. I'll add that.

Nice picture by the way ;)

Comment (elemoine) [reply lib/pc_patch_uncompressed.c]: New commit added.

**Pull request merged**